snappy_wrappers/wrappers/cbioportal/merge_tables/script.R

```diff
@@ -0,0 +1,585 @@
+require(magrittr)
+
+#' Read & merge sample files into cBioPortal gene-based data table.
+#'
+#' Used for expression tables, log2 CNA, pseudo-Gistic table.
+#'
+#' @param fns named list or vector of sample files, with sample ids as names
+#' @param mappings mapping table between feature ids (must contain ENSEMBL, SYMBOL & ENTREZ_ID columns)
+#' @param type type of data to merge
+#' @param args named list of additional arguments required to process the data before exporting
+#' @param remove_feature_version remove the trailing ".<number>" in the feature id (useful for GENCODE)
+#'
+#' @example
+merge_tables <- function(fns, mappings, type=c("log2", "gistic", "expression"), args=list(), remove_feature_version=TRUE) {
+    if (missing(fns) || is.null(fns))
+        stop("Missing list of sample files")
+    if (is.list(fns)) {
+        stopifnot(all(lengths(fns) == 1))
+        fns <- unlist(fns)
+    }
+    if (!is.character(fns) || any(is.na(fns)) || any(fns==""))
+        stop("Missing at least one sample filename")
+    if (is.null(names(fns)) || any(is.na(names(fns))) || any(names(fns)==""))
+        stop("Missing at least one sample id")
+
+    if (missing(mappings) || is.null(mappings) || !is.character(mappings) || length(mappings)!=1 || is.na(mappings) || mappings=="")
+        stop("Missing or illegal gene id mappings filename")
+    mappings <- get_id_mappings(mappings, verbose=TRUE)
+
+    if (type == "expression") {
+        stopifnot(all(c("pipeline_id", "tx_obj") %in% names(args)))
+        counts <- read_sample_files(fns, featureCol="V1", valueCol="V2", header=FALSE)
+        tmp <- compute_rpkm(counts, tx_obj=args$tx_obj, verbose=TRUE)
+        method <- "sum"
+    }
+
+    if (type == "gistic") {
```

Comment:
Are you creating also information for deletion?

Reply:
Yes, deletions and losses are also part of the final table. Here is what the new comments in the code say:

```
# Copy numbers (in "cn" column) are transformed into (pseudo-) gistic codes:
# 0: Deep deletion, 1: heterozygous deletion, 2: copy number neutral, 3: gain, 4: amplification
# In https://doi.org/10.1038/s41586-022-04738-6, the amplification is defined as 
# copy number greater or equal to 9 copies (args$amplification = 9).
```

I hope it is clearer now